components/bpel/org.wso2.carbon.bpel.ui/src/main/java/org/wso2/carbon/bpel/ui/bpel2svg/impl/ScopeImpl.java

```diff
@@ -691,4 +716,49 @@ protected Element getImageDefinition(SVGDocument doc, String imgPath, int imgXLe
 
         return group;
     }
+
+    //Different Implementations for start and end scope icons
+    @Override
+    protected Element getEndImageDefinition(SVGDocument doc) {
+        return getStartEndImageDef(doc, getEndIconPath(), getEndIconXLeft(),
+                getEndIconYTop(), getEndIconWidth(), getEndIconHeight(),
+                getEndImageId());
+    }
+
+    protected Element getStartImageDefinition(SVGDocument doc) {
+        return getStartEndImageDef(doc, getStartIconPath(), getStartIconXLeft(),
+                getStartIconYTop(), getStartIconWidth(), getStartIconHeight(),
+                getStartImageId());
+    }
+
+    protected Element getStartEndImageDef(SVGDocument doc, String imgPath, int imgXLeft, int imgYTop,
+                                          int imgWidth, int imgHeight, String id) {
+
+        Element group = null;
+        group = doc.createElementNS("http://www.w3.org/2000/svg", "g");
+        group.setAttributeNS(null, "id", getLayerId());
+
+        if (getStartIconPath() != null) {
+
+            Element x = null;
+            x = doc.createElementNS("http://www.w3.org/2000/svg", "g");
+            x.setAttributeNS(null, "id", id);
+
+            int embedImageX = imgXLeft + imgWidth;
+            int embedImageY = imgYTop + (5 / 2);
+
+            Element embedImage = doc.createElementNS("http://www.w3.org/2000/svg", "line");
+            embedImage.setAttributeNS(null, "x1", String.valueOf(imgXLeft + 10));
```

Comment:
Use constants for these strings. You can have a common class like "bpel2SvgCostants" and include all there.